wgsl/index.bs

```diff
@@ -2751,29 +2929,116 @@ ISSUE: Can we query upper bounds on workgroup size dimensions?  Is it independen
   </xmp>
 </div>
 
-## Shader Interface TODO ## {#shader-interface}
+## Shader Interface ## {#shader-interface}
+
+The shader interface is the set of objects
+through which the shader accesses data external to the shader stage,
+either for reading or writing.
+The interface includes:
+
+* Pipeline inputs and outputs
+* Buffer resources
+* Texture resources
+
+These objects are represented by module-scope variables in certain storage classes.
+
+We say a variable is <dfn noexport>statically accessed</dfn> by a function if any subexpression
+in the body of the function uses the variable's identifier,
+and that subexpression is [=in scope=] of the variable's declaration.
+Note that being statically accessed is independent of whether an execution of the shader
+will actually evaluate the subexpression, or even execute the enclosing statement.
+
+More precisely, the <dfn noexport>interface of a shader stage</dfn>
```

Comment:
Why are we introducing this definition? IIRC nothing on the host side would care about the combined interface of a particular shader stage in a module. It would care about the module as a whole, or particular entry points.

Or maybe this paragraph should talk about "interface of an entry point" rather than "interface of a shader stage".

Reply:
The "entry point" is a WGSL function that serves a specific role. 

The "shader stage" is the broader thing defined in 8.1 "Shader stages":  It's the entry point + all the data objects accessed by the entry point and the static call tree of the entry point, and is executed for a particular pipeline stage.

>  IIRC nothing on the host side would care about the combined interface of a particular shader stage in a module. 

You need to know what needs to be bound (binding points, and attachments). Also the programmer needs to know what data becomes accessible to the shader and when that data  is set up (e.g. supplying values to the pipeline inputs).